framework/i18n/GettextMessageSource.php

```diff
@@ -103,7 +103,11 @@ protected function loadFallbackMessages($category, $fallbackLanguage, $messages,
         $fallbackMessageFile = $this->getMessageFilePath($fallbackLanguage);
         $fallbackMessages = $this->loadMessagesFromFile($fallbackMessageFile, $category);
 
-        if ($messages === null && $fallbackMessages === null && $fallbackLanguage !== $this->sourceLanguage) {
+        if (
+            $messages === null && $fallbackMessages === null
+            && $fallbackLanguage !== $this->sourceLanguage
+            && $fallbackLanguage !== substr($this->sourceLanguage, 0, 2)
```

Comment:
Not sure it's correct. See https://github.com/yiisoft/yii2/pull/8057/files#diff-97dd491a4c8c30f7d804fbea50cd0b40R293